src/main/java/net/dv8tion/jda/api/entities/User.java

```diff
@@ -225,4 +228,168 @@ default String getEffectiveAvatarUrl()
+        STAFF(                     0, "Discord Employee"),
+        PARTNER(                   1, "Discord Partner"),
+        HYPESQUAD(                 2, "HypeSquad Events"),
+        BUG_HUNTER_LEVEL_1(        3, "Bug Hunter Level 1"),
+
+        // HypeSquad
+        HYPESQUAD_BRAVERY(         6, "HypeSquad Bravery"),
+        HYPESQUAD_BRILLIANCE(      7, "HypeSquad Brilliance"),
+        HYPESQUAD_BALANCE(         8, "HypeSquad Balance"),
+
+        EARLY_SUPPORTER(           9, "Early Supporter"),
+        TEAM_USER(                10, "Team User"),
+        SYSTEM(                   12, "System User"),
+        BUG_HUNTER_LEVEL_2(       14, "Bug Hunter Level 2"),
+        VERIFIED_BOT(             16, "Verified Bot"),
+        VERIFIED_DEVELOPER(       17, "Verified Bot Developer"),
+        
+        UNKNOWN(                   -1, "Unknown");
+
+        /**
+         * Empty array of Flag enum, useful for optimized use in {@link java.util.Collection#toArray(Object[])}.
+         */
+        public static final UserFlag[] EMPTY_FLAGS = new UserFlag[0];
+        
+        private final int offset;
+        private final long raw;
+        private final String name;
+
+        UserFlag(int offset, @Nonnull String name)
+        {
+            this.offset = offset;
+            this.raw = 1 << offset;
+            this.name = name;
+        }
+
+        /**
+         * The readable name as used in the Discord Client.
+         * 
+         * @return The readable name of this {@link net.dv8tion.jda.api.entities.User.UserFlag Flag}.
+         */
+        @Nonnull
+        public String getName()
+        {
+            return this.name;
+        }
+
+        /**
+         * The binary offset of the flag.
+         * 
+         * @return The offset that represents this {@link net.dv8tion.jda.api.entities.User.UserFlag Flag}.
+         */
+        public int getOffset()
+        {
+            return offset;
+        }
+
+        /**
+         * The value of this flag when viewed as raw value.
+         * <br>This is equivalent to: <code>1 {@literal <<} {@link #getOffset()}</code>
+         * 
+         * @return The raw value of this specific flag.
+         */
+        public long getRawValue()
+        {
+            return raw;
+        }
+
+        /**
+         * Gets the first {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} relating to the provided offset.
+         * <br>If there is no {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} that matches the provided offset,
+         * {@link net.dv8tion.jda.api.entities.User.UserFlag#UNKNOWN Flag.UNKNOWN} is returned.
+         * 
+         * @param  offset
+         *         The offset to match a {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} to.
+         *         
+         * @return {@link net.dv8tion.jda.api.entities.User.UserFlag Flag} relating to the provided offset.
+         */
+        @Nonnull
+        public static UserFlag getFromOffset(int offset)
```

Comment:
I wonder what a valid use-case for this would be. When would you work with the raw offsets in client-code? Isn't that exposing the implementation a bit too much?

What I'm saying is: if you're implementing a `UserFlag` enum to encapsulate the `public_flag` numbers, why would you allow the client code to still work with the numbers? Seems a bit off to me, I'd like to hear your thoughts

Reply:
Most of the code is based on the `Permission` enum, which also has a `getFromOffset` method for this.